### PR TITLE
[AIRFLOW-4449] updated default permission for custom roles to 'Viewer'

### PR DESCRIPTION
**Pull request by kanetkarster**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
- https://issues.apache.org/jira/browse/AIRFLOW-4449

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Previously, it was impossible to create a custom role with permissions
less than viewer. This was suboptimal if, for example, we wanted to
create a user without the `set_success` permission

This also makes the behavior of the mode match the associated comment.\

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: it's a pretty trivial change and is already covered by `test_init_role_baseview`

manually tested that the web interface behaved as expected

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by kanetkarster** [comment]
cc @feng-tao - this follows up from your change in #3197

File: airflow/www/security.py [hunk 1]
```diff
@@ -416,7 +416,7 @@ def merge_pv(perm, view_menu):
         # for all the dag-level role, add the permission of viewer
```
Comment (kanetkarster) [inline airflow/www/security.py]: this comment is now up-to-date

**Comment by kanetkarster** [comment]
I just wanted to bump this again; cc @feng-tao

**Comment by ashb** [comment]
Can you explain in a bit more detail what this change does, and what effect it will have?

**Comment by kanetkarster** [comment]
Right now, all custom roles have, as a base, all the permissions given to User. Whenever `create_custom_dag_permission_view` get's called, the permissions of custom roles get updated.

We want to create Roles with permissions not included with in User. In particular, we don't want all roles to have access to the `set success on _____` permissions.

This change will update that default permissions given to custom roles to be Viewer, so we can assign "write" permissions to roles as they are needed

FWIW, I think once #4960 is included in the release, there will be better ways to get around this. Right now, we've configured our custom SecurityManager class to not call `create_custom_dag_permission_view` ever, which is less than desirable
Regardless, I think it's a saner default for all custom roles to have no write permissions

**Comment by OmerJog** [comment]
@ashb ping :)

File: airflow/www/security.py [hunk 2]
```diff
@@ -416,7 +416,7 @@ def merge_pv(perm, view_menu):
...
         # with the dag view to ab_permission_view
         all_roles = self.get_all_roles()
-        user_role = self.find_role('User')
+        viewer_role = self.find_role('Viewer')
```
Comment (ashb) [inline airflow/www/security.py]: Please make the "default" role a config option under the `webserver` section.

**Comment by OmerJog** [comment]
@kanetkarster are you working on this pr?

**Comment by kanetkarster** [comment]
Lost track of this, I had made the update but forgot to push. Will get that
done tomorrow

File: airflow/www/security.py [hunk 3]
```diff
@@ -27,7 +27,7 @@
 from airflow.exceptions import AirflowException
 from airflow.utils.db import provide_session
 from airflow.utils.log.logging_mixin import LoggingMixin
-from airflow.www.app import appbuilder
+from airflow.www.app import app, appbuilder
```
Comment (kanetkarster) [inline airflow/www/security.py:30]: I had initially used appbuilder, like was done in `get_user_roles` but that caused all of the tests to fail, so I started directly using `app.config`. Is there something I might have been missing?

**Pull request closed without merging**